MCA/PS2/PS2Application.cpp

```diff
@@ -113,6 +112,34 @@ int CPS2Application::main(int argc, char *argv[])
 	return 0;
 }
 
+void CPS2Application::initLanguage()
+{
+	std::string langfile = CResources::boot_path + "lang.lng";
+	if (!loadLanguage(langfile))
+	{
+		int systemLanguage = configGetLanguage();
+		static const char* languageFiles[] = {
+			"lang.lng",   // Japanese has no support in the font
+			"lang_en.lng",
+			"lang_fr.lng",
+			"lang_es.lng",
+			"lang_de.lng",
+			"lang_it.lng",
+			"lang_du.lng",
+			"lang_pt.lng",
+			"lang_ru.lng", // Requires XEB+ 2024 onwards to be detected, else, it will default to 1 (English)
+			"lang.lng",   // Korean has no support in the font
+			"lang.lng",	// Chinese has no support in the font
+			"lang.lng"	// Chinese has no support in the font
+		};
+		if (systemLanguage >= 1 && systemLanguage <= 8)
+		{
+			std::string langfile = CResources::boot_path + languageFiles[systemLanguage];
+			loadLanguage(langfile);
+		}
+	}
+}
```

Comment:
Move the array to the top to make control flow more readable, remove unused languages at the end and comment that. Also you're re-declaring langfile, which is not an error, but not a good practice either. Check suggestion.
```suggestion
void CPS2Application::initLanguage()
{
	static const char* languageFiles[] = {
		"lang.lng",   // Japanese has no support in the font
		"lang_en.lng",
		"lang_fr.lng",
		"lang_es.lng",
		"lang_de.lng",
		"lang_it.lng",
		"lang_du.lng",
		"lang_pt.lng",
		"lang_ru.lng" // Requires XEB+ 2024 onwards to be detected, else, it will default to 1 (English)
	};
	
	std::string defaultLanguageFile = CResources::boot_path + "lang.lng";
	if (!loadLanguage(defaultLanguageFile))
	{
		int systemLanguage = configGetLanguage();
		if (systemLanguage >= 1 && systemLanguage <= 8) //Korean and Chinese languages do not have a valid font yet
		{
			std::string systemLanguageFile = CResources::boot_path + languageFiles[systemLanguage];
			loadLanguage(systemLanguageFile);
		}
	}
}
```